static/app/actionCreators/metrics.tsx

```diff
@@ -69,3 +78,46 @@ export const doMetricsRequest = (
 
   return api.requestPromise(pathname, {includeAllArgs, query: urlQuery});
 };
+
+export function fetchMetricsTags(
+  api: Client,
+  orgSlug: Organization['slug'],
+  projects?: number[]
+): Promise<MetricTag[]> {
```

Comment:
Not a blocker, but this function should also accept an array of selected fields. 
API will then return an intersection of tags available for each of the metrics.